parallel/client.py

```diff
@@ -93,6 +94,10 @@ class CUDAPoolClient(ProcessPoolClient):
     will run on a different GPU modulo num workers, which should be set to
     the number of GPUs.
     """
+    def __init__(self, max_workers):
+        super().__init__(max_workers)
+        if self.max_workers > get_gpu_count():
```

Comment:
I ran into a case where things die if you have workers greater than GPU count, so this seemed helpful to me though maybe it could be noisy in some situation?

Reply:
I would probably assert in this case. The code probably wouldn't even run as it'd be setting `CUDA_VISIBLE_DEVICES` to garbage.